ddp-workspace/projects/ddp-sdk/src/lib/utility/RegExpHelper.ts

```diff
@@ -0,0 +1,50 @@
+export class RegExpHelper {
+
+    /* The method returns a RegExp:
+    *  if text includes symbols to substitute, they match with any of `substitutableSymbols` or `replaceValue`(a space by default),
+    *  (case/space insensitive)
+    *  e.g.
+    *  regExp = createSubstitutableSymbolsRegExp('asd / zxc', ['-', '/'], ' ');
+    *  regExp.test('asd zxc') = true;
+    *  regExp.test('asd / zxc') = true;
+    *  regExp.test('asd -    zxc') = true;
+    */
+    static createSubstitutableSymbolsRegExp(text: string, substitutableSymbols: string[] = [], replaceValue = ' '): RegExp {
+        const anyAmountOfSpacesRegExp = '\\s*';
+        if (substitutableSymbols.includes(text) || !substitutableSymbols.some(s => text.includes(s))) {
+            // return same text but space/case insensitive
+            return new RegExp(
+                RegExpHelper.replaceEmptySpaces(
+                    RegExpHelper.escapeRegExp(text),
+                    anyAmountOfSpacesRegExp
+                ),
+                'gi'
+            );
+        }
+
+        const escapedSubstitutableSymbols = substitutableSymbols.map(symbol => RegExpHelper.escapeRegExp(symbol));
+        const substitutableSymbolsRegExp = new RegExp(escapedSubstitutableSymbols.join('|'), 'g');
+
+        const pattern = RegExpHelper.escapeRegExp(
+                RegExpHelper.replaceEmptySpaces(text, ''),
```

Comment:
do we want to remove all spaces between words? not sure it would work if the option has multiple words + substitutable symbols

Reply:
'brain stem glioma d' string doesn't find 'Brain stem glioma / Diffuse Intrinsic Pontine Glioma (DIPG)'
maybe it's related

Reply:
The example above should not be working (from required behavior).
e.g.
`B-cell` should match `B cell`
`B/cell` should match `B cell`
but
`B cell` should not match neither `B-cell` nor `B/cell`

Reply:
Actually, what we are looking for here is is to include more matches, not less.
So if option label is B-Cell
And user enters B Cell or B-Cell or B/Cell or b-cell or B-CELL or B CELL, etc, I want to see the existing option B-Cell.
And same with the reverse: i.e., if we have B Cell in option list, I want to a match with all these weird combinations.
I think this can be achieved without regex (not saying not to use regex, just saying that the matching we are talking about does not require it).
So it would looks like something like:
1. Normalize all the options labels, converting everything to uppercase and replace all "substitutable" symbols to spaces OR convert the labels to list of words.
2. Apply same conversion to the entered search query.
3. Look for matches.

One thing to emphasize is that the ticket is more about addressing CMI's generic concern that users will enter something and they will miss some of the cancer options that ARE in the option list, but because they used or did not use a given separator or used a given capitalization or even the word order is different, user did not get a match.

I would add that as I look at the cancer list, we should expand functionality here to have words to ignore, e.g., I would like configure the cancer list so that the word "cancer" is ignored.

I take responsibility for not speccing this is in greater detail. But at same time, I am asking for engineer to put forth a solution that addresses the user's concern.